src/components/layers/basemaps/BasemapList.js

```diff
@@ -1,26 +1,29 @@
+import { useCachedDataQuery } from '@dhis2/analytics'
 import PropTypes from 'prop-types'
 import React from 'react'
 import { layerTypes } from '../../map/MapApi.js'
 import Basemap from './Basemap.js'
 import styles from './styles/BasemapList.module.css'
 
-const BasemapList = ({ selectedID, basemaps, selectBasemap }) => (
-    <div className={styles.basemapList} data-test="basemaplist">
-        {basemaps
-            .filter((basemap) => layerTypes.includes(basemap.config.type))
-            .map((basemap, index) => (
-                <Basemap
-                    key={`basemap-${index}`}
-                    onClick={selectBasemap}
-                    isSelected={basemap.id === selectedID}
-                    {...basemap}
-                />
-            ))}
-    </div>
-)
+const BasemapList = ({ selectedID, selectBasemap }) => {
+    const { basemaps } = useCachedDataQuery()
+    return (
+        <div className={styles.basemapList} data-test="basemaplist">
+            {basemaps
+                .filter((basemap) => layerTypes.includes(basemap.config.type))
```

Comment:
@turban Is there a reason why we shouldn't just do this filtering at the very beginning when the basemap list is first created? here: https://github.com/dhis2/maps-app/pull/2940/files#diff-4c1bfe25cf9d54e82006ccbc93b64f4b14e5e60dc2139eaf3c9c438671498117R30

As in, are there cases where we need the basemaps that dont have a `type`? Maybe for backwards compatibility where old maps are using these basemaps...